pdns/pdns_recursor.cc

```diff
@@ -1118,6 +1118,13 @@ static void startDoResolve(void *p)
     // Check if the query has a policy attached to it
     if (wantsRPZ) {
       appliedPolicy = luaconfsLocal->dfe.getQueryPolicy(dc->d_mdp.d_qname, dc->d_source, sr.d_discardedPolicies);
+
```

Comment:
Note that you will only record the `RPZ` hits on the `qname` and `source address`, not the ones on the authoritative servers queried during the resolving process, nor the ones based on the content of the response.

Reply:
It also misses any changes that can happen in pre/post resolve. I will put more thought into this. It looks like it might need to be handled in both the drop spots, and then down in ```haveAnswer``` after all hooks are done.